hardware_interface/src/resource_manager.cpp

```diff
@@ -195,6 +197,58 @@ class ResourceStorage
     return result;
   }
 
+  void remove_all_hardware_interfaces_from_available_list(const std::string & hardware_name)
+  {
+    // remove all command interfaces from available list
+    for (const auto & interface : hardware_info_map_[hardware_name].command_interfaces)
+    {
+      auto found_it = std::find(
+        available_command_interfaces_.begin(), available_command_interfaces_.end(), interface);
```

Comment:
this is a search in a for loop... could this be implemented somehow using a more flat mapping by any chance? please feel free to say that I'm asking for impossible or very tedious things

Reply:
I will check this after I address above two comments that could also influence this.